### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by Daedalusspacegames** [aristocratos/btop#441]
**Read the README.md and search for similar issues before posting a bug report!**

Any bug that can be solved by just reading the [prerequisites](https://github.com/aristocratos/btop#prerequisites) section of the README will likely be ignored.

**Describe the bug**

Starting with btop-- version 1.2.10, the number of active CPUs displayed is incorrect

**To Reproduce**

1. Install btop using official x86_64 tar.gz
2. Run btop
3. CPU count will display an incorrect number of CPUs (in my case 32, or 20 more)

**Expected behavior**

My Ryzen 5 1600X has 6 cores and hyperthreading, so the CPU graph should display 12 CPUs (C0 to C11).

**Screenshots**

[Screenshot](https://media.discordapp.net/attachments/700957087232163840/1027885045601554482/unknown.png)

**Info (please complete the following information):**
 - btop++ version: 1.2.10
 - Binary: Static binary from release (installed via AUR)
 - Architecture: x86_64
 - Platform: Linux
 - (Linux) Kernel: 6.0.0-270-tkg-cfs
 - Terminal used: Kitty (Konsole also tested)
 - Font used: TerminessTTF Nerd Font Mono

**Additional context**

```
2022/10/07 (03:13:28) | ===> btop++ v.1.2.10
2022/10/07 (03:13:28) | DEBUG: Starting in DEBUG mode!
2022/10/07 (03:13:28) | INFO: Logger set to DEBUG
2022/10/07 (03:13:28) | DEBUG: Setting LC_ALL=en_US.UTF-8
2022/10/07 (03:13:28) | INFO: Running on /dev/pts/0
2022/10/07 (03:13:28) | DEBUG: Loading theme file: /usr/share/btop/themes/dusklight.theme
```

**GDB Backtrace**

If btop++ is crashing at start the following steps could be helpful:

(Extra helpful if compiled with `make OPTFLAGS="-O0 -g"`)

1. run (linux): `gdb btop` (macos): `lldb btop`

2. `r` to run, wait for crash and press enter if prompted, CTRL+L to clear screen if needed.

4. (gdb): `thread apply all bt` (lldb): `bt all` to get backtrace for all threads

5. Copy and paste the backtrace here:

**Comment by Daedalusspacegames** [comment]
For the record, this issue also occurred when compiling btop++ from source with GCC 11.3.0. In both cases the backtrace produced no output.

**Comment by aristocratos** [comment]
@Daedalusspacegames 
> CPU count will display an incorrect number of CPUs (in my case 32, or 20 more)

Is it 32 or 20 more? Because the value used for setting number of cores is the number of cores the kernel reports it has configured.

You can try the attached binary which uses another method and see if it works better:
[btop-musl-x64-test.zip](https://github.com/aristocratos/btop/files/9733426/btop-musl-x64-test.zip)

**Comment by Daedalusspacegames** [comment]
32 total. 20 more than the real value, which is 12.

The attached binary prints the following error:
```
ERROR: No UTF-8 locale detected!
Use --utf-force argument to forcee start if you're sure your terminal can handle it
```
However, running the binary with --utf-force or with the environment variable LC_ALL=C.UTF-8 works. With this binary, the issue appears to be solved.

**Comment by aristocratos** [comment]
Fixed in [v1.2.11](https://github.com/aristocratos/btop/releases/tag/v1.2.11)
